alchemy/src/cloudflare/do-state-store/store.ts

```diff
@@ -0,0 +1,293 @@
+    return Object.fromEntries(
+      await Promise.all(
+        Object.entries(result).map(async ([key, value]) => [
+          key,
+          await deserializeState(this.scope, value),
+        ]),
+      ),
+    );
+  }
+
+  async all(): Promise<Record<string, State>> {
+    const client = await this.getClient();
+    const res = await client.rpc("all");
+    if (!res.ok) {
+      return {};
+    }
+    const result: Record<string, string> = await res.json();
+    return Object.fromEntries(
+      await Promise.all(
+        Object.entries(result).map(async ([key, value]) => [
+          key,
+          await deserializeState(this.scope, value),
+        ]),
+      ),
+    );
+  }
+
+  async set(key: string, value: State): Promise<void> {
+    const client = await this.getClient();
+    const serialized = await serialize(this.scope, value);
+    await client.rpc("set", { key }, serialized);
+  }
+
+  async delete(key: string): Promise<void> {
+    const client = await this.getClient();
+    await client.rpc("delete", { key });
+  }
+
+  async list(): Promise<string[]> {
+    const client = await this.getClient();
+    const res = await client.rpc("list");
+    return await res.json();
+  }
+
+  async count(): Promise<number> {
+    const client = await this.getClient();
+    const res = await client.rpc("count");
+    return await res.json();
+  }
+
+  private async getClient(): Promise<DOFetchClient> {
+    this.client ??= this.createClient();
+    return await this.client;
+  }
+
+  private async createClient(): Promise<DOFetchClient> {
+    const workerName = this.options.worker?.name ?? "alchemy-state";
+    const token = this.options.worker?.token ?? process.env.ALCHEMY_STATE_TOKEN;
+    if (!token) {
+      throw new Error(
+        "DOStateStore requires an API key. Please set options.worker.token or the ALCHEMY_STATE_TOKEN environment variable.",
+      );
+    }
+    if (this.options.worker?.url) {
+      const client = new DOFetchClient(
+        this.options.worker.url,
+        token,
+        this.prefix,
+      );
+      const res = await client.fetch("/status");
+      if (res.ok) {
+        return client;
+      }
+      if (res.status === 401) {
+        throw new Error(
+          "A worker URL was provided to DOStateStore, but the token is incorrect. Correct the token or remove the worker URL to create a new worker.",
+        );
+      }
+      throw new Error(
+        `A worker URL was provided to DOStateStore, but the worker status is ${res.status} ${res.statusText}.`,
+      );
+    }
+    const api = await createCloudflareApi(this.options);
+    const [subdomain, worker] = await Promise.all([
+      this.getAccountSubdomain(api),
+      getWorkerScriptMetadata(api, workerName),
+    ]);
+    const client = new DOFetchClient(
+      `https://${workerName}.${subdomain}.workers.dev`,
+      token,
+      this.prefix,
+    );
+    if (worker && !this.options.worker?.force) {
+      return client;
+    }
```

Comment:
what does force mean?

Reply:
Force replace the worker if it already exists, mainly for debugging. I'll add docs+comments to make this clearer.

I'm also thinking we can use the tag for versioning so if we want to update the worker, we can do that. In those cases, you wouldn't need to force.